docs/proposals/2017-5-26 - Advanced Table.md

```diff
@@ -0,0 +1,446 @@
+# Problem
+
+Current ANS tables are not sufficient to support tables from methode.
+
+
+# Proposal
+
+Add a new table object that encapsulates the functionality of an HTML table, called `advanced_table`.
+
+Key traits of the advanced table:
+
+* made up of 4 objects - `advanced_table`, `row`, `data_cell`, `header_cell`
+  * a `table` is a list of rows
+  * a `row` is a list of cells
+  * a `cell` is a list of content elements, restricted to text and raw_html
+    * there are two implementations of a cell, `data_cell` and `header_cell`, similar to `<th>` and `<td>` tags
+* order/layout assumptions - elements will have an assumed layout that clients may follow
+  * for example, rows are ordered from top to bottom, and cells are ordered left to right
+* no other rendering details will be included in the spec
+  * if needed, adapters can use additional properties to specify rendering details
+
+
+## Advanced Table
+
+`advanced_table` is the parent object. It is a content element, and contain a list of rows.
+
+### Rows
+
+List of rows that belong to the table. The order is assumed to be top to bottom.
+
+### Title
+
+Text field. Name of the table.
+
+### Citation
+
+ANS text element. The source of the table, for example "The Washington Post".
+
+### Credits
+
+ANS credits object, for identifying the author(s) of the table.
+
+### Subtype
+
+ANS subtype. Can be used to distinguish between different types of tables, so that clients can react accordingly.
+
+### No height/width or other display options
+
+These are intentionally left out.
+
+If adapters need to include any style or rendering details, they can use additional properties on any component of a table - table, row, cell, or content element.
+
```

Comment:
This is probably the strongest reason for us to go the route of raw_html. I don't think we will be able to get around this problem without including a significant amount of rendering specification in the spec.